scripts/run_acceptance_tests.py

```diff
@@ -0,0 +1,270 @@
+
+from scripts import common  # isort:skip pylint: disable=wrong-import-position, unused-import
+
+from core.constants import constants  # isort:skip
+from scripts import build  # isort:skip
+from scripts import flake_checker  # isort:skip
+from scripts import install_third_party_libs  # isort:skip
+from scripts import servers  # isort:skip
+
+MAX_RETRY_COUNT: Final = 3
+GOOGLE_APP_ENGINE_PORT: Final = 8181
+ELASTICSEARCH_SERVER_PORT: Final = 9200
+PORTS_USED_BY_OPPIA_PROCESSES: Final = [
+    GOOGLE_APP_ENGINE_PORT,
+    ELASTICSEARCH_SERVER_PORT,
+]
+
+_PARSER: Final = argparse.ArgumentParser(
+    description="""
+Run this script from the oppia root folder:
+   python -m scripts.run_acceptance_tests
+
+The root folder MUST be named 'oppia'.
+""")
+
+_PARSER.add_argument(
+    '--skip-install',
+    help='If true, skips installing dependencies. The default value is false.',
+    action='store_true')
+_PARSER.add_argument(
+    '--skip-build',
+    help='If true, skips building files. The default value is false.',
+    action='store_true')
+_PARSER.add_argument(
+    '--sharding-instances', type=int, default=3,
+    help='Sets the number of parallel browsers to open while sharding. '
+         'Sharding must be disabled (either by passing in false to --sharding '
+         'or 1 to --sharding-instances) if running any tests in isolation '
+         '(fit or fdescribe).')
+_PARSER.add_argument(
+    '--prod_env',
+    help='Run the tests in prod mode. Static resources are served from '
+         'build directory and use cache slugs.',
+    action='store_true')
+_PARSER.add_argument(
+    '--suite', default='full',
+    help='Performs test for different suites'
+         'For performing a full test, no argument is required.')
+_PARSER.add_argument(
+    '--chrome_driver_version',
+    help='Uses the specified version of the chrome driver')
+_PARSER.add_argument(
+    '--server_log_level',
+    help='Sets the log level for the appengine server. The default value is '
+         'set to error.',
+    default='error',
+    choices=['critical', 'error', 'warning', 'info'])
+_PARSER.add_argument(
+    '--source_maps',
+    help='Build webpack with source maps.',
+    action='store_true')
+_PARSER.add_argument(
+    '--mobile',
+    help='Run e2e test in mobile viewport.',
+    action='store_true')
+
+
+def is_oppia_server_already_running() -> bool:
+    """Check if the ports are taken by any other processes. If any one of
+    them is taken, it may indicate there is already one Oppia instance running.
+
+    Returns:
+        bool. Whether there is a running Oppia instance.
+    """
+    for port in PORTS_USED_BY_OPPIA_PROCESSES:
+        if common.is_port_in_use(port):
+            print(
+                'There is already a server running on localhost:%s. '
+                'Please terminate it before running the end-to-end tests. '
+                'Exiting.' % port)
+            return True
+    return False
```

Comment:
This is copy of the same function in e2e test runner, please move it to common to remove redundancy.

Reply:
ditto for other functions here

Reply:
Sure ! Will do it